### ISSUE TITLE
Master branch is failing

### ISSUE THREAD

**Issue by nzakas** [eslint/eslint#2001]
When running the tests, there's a failure related to destructuring:

```
  1) eslint Edge cases should not crash when parsing destructured assignment:
     Error: Unknown node type AssignmentPattern.
      at Controller.traverse (c:\Users\Nicholas\projects\personal\jscheck\node_m
odules\escope\node_modules\estraverse\estraverse.js:517:31)
      at Object.traverse (c:\Users\Nicholas\projects\personal\jscheck\node_modul
es\escope\node_modules\estraverse\estraverse.js:709:27)
      at traverseIdentifierInPattern (c:\Users\Nicholas\projects\personal\jschec
k\node_modules\escope\escope.js:886:20)
      at Referencer.extend.visitPattern (c:\Users\Nicholas\projects\personal\jsc
heck\node_modules\escope\escope.js:1034:13)
      at Referencer.extend.visitVariableDeclaration (c:\Users\Nicholas\projects\
personal\jscheck\node_modules\escope\escope.js:1178:18)
      at Referencer.extend.VariableDeclaration (c:\Users\Nicholas\projects\perso
nal\jscheck\node_modules\escope\escope.js:1346:22)
      at Referencer.Visitor.visit (c:\Users\Nicholas\projects\personal\jscheck\n
ode_modules\escope\node_modules\esrecurse\esrecurse.js:109:34)
      at Referencer.Visitor.visitChildren (c:\Users\Nicholas\projects\personal\j
scheck\node_modules\escope\node_modules\esrecurse\esrecurse.js:88:38)
      at Referencer.extend.Program (c:\Users\Nicholas\projects\personal\jscheck\
node_modules\escope\escope.js:1246:18)
      at Referencer.Visitor.visit (c:\Users\Nicholas\projects\personal\jscheck\n
ode_modules\escope\node_modules\esrecurse\esrecurse.js:109:34)
      at Object.analyze (c:\Users\Nicholas\projects\personal\jscheck\node_module
s\escope\escope.js:1440:20)
      at EventEmitter.module.exports.api.verify (c:\Users\Nicholas\projects\pers
onal\jscheck\lib\eslint.js:9:25112)
      at Context.<anonymous> (c:\Users\Nicholas\projects\personal\jscheck\tests\
lib\eslint.js:2226:20)
      at callFn (c:\Users\Nicholas\projects\personal\jscheck\node_modules\mocha\
lib\runnable.js:251:21)
      at Test.Runnable.run (c:\Users\Nicholas\projects\personal\jscheck\node_mod
ules\mocha\lib\runnable.js:244:7)
      at Runner.runTest (c:\Users\Nicholas\projects\personal\jscheck\node_module
s\mocha\lib\runner.js:374:10)
      at c:\Users\Nicholas\projects\personal\jscheck\node_modules\mocha\lib\runn
er.js:452:12
      at next (c:\Users\Nicholas\projects\personal\jscheck\node_modules\mocha\li
b\runner.js:299:14)
      at c:\Users\Nicholas\projects\personal\jscheck\node_modules\mocha\lib\runn
er.js:309:7
      at next (c:\Users\Nicholas\projects\personal\jscheck\node_modules\mocha\li
b\runner.js:248:23)
      at Object._onImmediate (c:\Users\Nicholas\projects\personal\jscheck\node_m
odules\mocha\lib\runner.js:276:5)
      at processImmediate [as _immediateCallback] (timers.js:345:15)
```

This failure appears to be coming from escope, as it doesn't recognize `AssignmentPattern`.

@Constellation

**Comment by nzakas** [comment]
We are using escope 2.0.7 and this is the code causing the problem:

```
var { a='a' } = {};
```

**Comment by nzakas** [comment]
Confirmed that `estraverse` in ESLint has a key for `AssignmentPattern`, so I'm not sure what the problem is here. I'm downgrading `escope` to unblock master.

**Comment by nzakas** [comment]
This was fixed through #2235.
